tests/Neo.UnitTests/SmartContract/UT_ApplicationEngine.cs

```diff
@@ -103,5 +106,103 @@ public void TestCheckingHardfork()
                 (setting[sortedHardforks[i]] > setting[sortedHardforks[i + 1]]).Should().Be(false);
             }
         }
+
+        [TestMethod]
+        public void TestSystem_Contract_Call_Permissions()
+        {
+            UInt160 scriptHash;
+            var snapshot = TestBlockchain.GetTestSnapshot();
+
+            // Setup: put a simple contract to the storage.
+            using (var script = new ScriptBuilder())
+            {
+                // Push True on stack and return.
+                script.EmitPush(true);
+                script.Emit(OpCode.RET);
+
+                // Mock contract and put it to the Managemant's storage.
+                scriptHash = script.ToArray().ToScriptHash();
+
+                snapshot.DeleteContract(scriptHash);
```

Comment:
why delete?

Reply:
It's just a cleanup, ensure that no such contract already exist in the storage before the test starts. This behaviour is nested from already existing test:
https://github.com/neo-project/neo/blob/fd1edf0ef20e80f2e22f2746d10b404920ca6461/tests/Neo.UnitTests/SmartContract/UT_InteropService.cs#L58

Reply:
I remember this type of problems in the past, to cleanup...I thought they were fixed.